docs/2-Interviews/answering_technical_questions.md

```diff
@@ -1,39 +1,129 @@
+  - There's a natural fear of being judged for saying something wrong, but it's often fine to keep talking even if you're not sure that you're right. Interviewers often want to know how you struggle just as much as (if not more than) they want to see you just "get it"
+  - One way to practice this is to do problems on a sheet of paper and [talk to a rubber duck](https://en.wikipedia.org/wiki/Rubber_duck_debugging)
+- Keep a good pace (keep in mind how long you have and how much you're aiming to get done ... but not too much cause it can be sort of terrifying)
+- It's definitely not about getting the right answer right away - any good coder will eventually write code to do it - it's about the process that got you there and getting there within time constraints
+- Make sure you handle common edge cases, like an empty input! People always forget that
+
+A common question students ask is **"am I allowed to use library functions?"** (e.g. find element in array). The answer is **probably yes**. However:
+- If the library solves the question in one line, the interviewer probably wants you to implement it to see if you understand how it works. Feel free to mention that an implementation already exist, but just don't start complaining about how you wouldn't do it in real-life (it's not the point)
+- Do explain very clearly the input and outputs of the function you are using, and when possible how it's implemented or what kind of tricky behavior it might have
+- Just to make sure, it's worth asking permission, or at least make sure the interviewer knows what the function is
+- It's fine to make up a function if you can't remember what the existing one is called. Nobody cares if you use `File.open(...)` when it's supposed to be `Sys.open_file(...)`
+
+On that subject, a very useful bit of preparation that can only take an hour or two is to familiarize yourself with the libraries that come with your language. It's worth reading over the operations you can do on strings, arrays/lists and maps/dictionaries. Useful functions include `zip`, `split`, `substr`/`slice`, `join`, `find`, etc.
+
+### Algorithms: finishing the interview
+
+- Once you have an initial implementation mostly right, ask if any further optimizations are necessary
+  - Are you optimizing for time? Space? Which is more important?
+- Offer to make further improvements the code. This tends to impress interviewers
+  - How can the style of your code be improved?
+  - Is there any redudancy that could be eliminated?
+  - Could you refactor your solution to be more generalized? e.g. accept an arbitrary iterable data structure as an input when the question just asked you to handle arrays
+  - Can you write the solution more concisely? e.g. you used a bunch of for loops when you could have used map/filter/reduce
+- Write/talk through some test cases if you have time, to prove to yourself and the interviewer that your code works
```

Comment:
I'd put this first on this list, I think it's more important to do than the other things